tensorboard/webapp/reloader/reloader_component_test.ts

```diff
@@ -47,6 +62,11 @@ describe('reloader_component', () => {
     dispatchSpy = spyOn(store, 'dispatch');
   });
 
+  function simulateVisibilityChange(visible: boolean) {
+    fakeDocument.visibilityState = visible ? 'visible' : 'hidden';
```

Comment:
do note that you are modifying the object over and over and it can lead to flaky test depending on random seed. Better way would be for us to create a new object on each `beforeEach`. 

```ts
function createFakeDocument() {
  return {...}
}

let document: DOCUMENT;

beforeEach(async () => {
  await TestBed.configureTestingModule({
      providers: [
        {
          provide: DOCUMENT,
          useFactory: createFakeDocument
        },
        ...
  );
  document = TestBed.get(Document);
});

```

Reply:
Done. But didn't type fakeDocument as Document since it would require additional hoops to fake visibilityState.